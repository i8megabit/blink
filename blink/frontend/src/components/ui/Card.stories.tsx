import type { Meta, StoryObj } from '@storybook/react';
import { Card, CardHeader, CardTitle, CardDescription, CardContent, CardFooter } from './Card';

const meta: Meta<typeof Card> = {
  title: 'UI/Card',
  component: Card,
  tags: ['autodocs'],
};
export default meta;

type Story = StoryObj<typeof Card>;

export const Basic: Story = {
  render: () => <Card className="p-4">Basic Card</Card>,
};

export const WithHeader: Story = {
  render: () => (
    <Card>
      <CardHeader>
        <CardTitle>Card Title</CardTitle>
        <CardDescription>Card description goes here</CardDescription>
      </CardHeader>
      <CardContent>
        <p>Card content goes here</p>
      </CardContent>
      <CardFooter>
        <button className="btn btn-primary">Action</button>
      </CardFooter>
    </Card>
  ),
};

export const OnlyContent: Story = {
  render: () => (
    <Card>
      <CardContent>
        <p>Only content inside Card</p>
      </CardContent>
    </Card>
  ),
};

export const WithFooter: Story = {
  render: () => (
    <Card>
      <CardContent>
        <p>Card with footer</p>
      </CardContent>
      <CardFooter>
        <button className="btn btn-secondary">Secondary</button>
      </CardFooter>
    </Card>
  ),
}; 