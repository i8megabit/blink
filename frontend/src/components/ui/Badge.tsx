import React from 'react';
import { cn } from '../../utils/cn';

interface BadgeProps extends React.HTMLAttributes<HTMLDivElement> {
  variant?: 'default' | 'secondary' | 'destructive' | 'outline' | 'success' | 'error' | 'warning';
  children: React.ReactNode;
}

export const Badge: React.FC<BadgeProps> = ({
  variant = 'default',
  className,
  children,
  ...props
}) => {
  const variants = {
    default: 'bg-primary text-primary-foreground hover:bg-primary/80',
    secondary: 'bg-secondary text-secondary-foreground hover:bg-secondary/80',
    destructive: 'bg-destructive text-destructive-foreground',
    outline: 'text-foreground border border-input',
    success: 'bg-green-100 text-green-800 border border-green-200',
    error: 'bg-red-100 text-red-800 border border-red-200',
    warning: 'bg-yellow-100 text-yellow-800 border border-yellow-200'
  };

  return (
    <div
      className={cn(
        'inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2',
        variants[variant],
        className
      )}
      {...props}
    >
      {children}
    </div>
  );
}; 