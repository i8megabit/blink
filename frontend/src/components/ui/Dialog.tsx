import * as React from "react"
import { X } from "lucide-react"
import { cn } from "../../lib/utils"

interface DialogProps {
  open?: boolean
  onOpenChange?: (open: boolean) => void
  children: React.ReactNode
}

interface DialogTriggerProps {
  asChild?: boolean
  children: React.ReactNode
}

interface DialogContentProps {
  children: React.ReactNode
  className?: string
}

interface DialogHeaderProps {
  children: React.ReactNode
  className?: string
}

interface DialogTitleProps {
  children: React.ReactNode
  className?: string
}

const DialogContext = React.createContext<{
  open: boolean
  onOpenChange: (open: boolean) => void
} | null>(null)

const Dialog: React.FC<DialogProps> = ({ open, onOpenChange, children }) => {
  return (
    <DialogContext.Provider value={{ open: open || false, onOpenChange: onOpenChange || (() => {}) }}>
      {children}
    </DialogContext.Provider>
  )
}

const DialogTrigger: React.FC<DialogTriggerProps> = ({ children }) => {
  const context = React.useContext(DialogContext)
  if (!context) throw new Error("DialogTrigger must be used within Dialog")

  return (
    <div onClick={() => context.onOpenChange(true)}>
      {children}
    </div>
  )
}

const DialogContent: React.FC<DialogContentProps> = ({ children, className }) => {
  const context = React.useContext(DialogContext)
  if (!context) throw new Error("DialogContent must be used within Dialog")

  if (!context.open) return null

  return (
    <div className="fixed inset-0 z-50 flex items-center justify-center">
      <div 
        className="fixed inset-0 bg-black bg-opacity-50 backdrop-blur-sm"
        onClick={() => context.onOpenChange(false)}
      />
      <div className={cn(
        "relative z-50 w-full max-w-lg rounded-lg border bg-white p-6 shadow-lg dark:bg-gray-800 dark:border-gray-700",
        className
      )}>
        {children}
        <button
          className="absolute right-4 top-4 rounded-sm opacity-70 transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:ring-offset-2"
          onClick={() => context.onOpenChange(false)}
        >
          <X className="h-4 w-4" />
          <span className="sr-only">Close</span>
        </button>
      </div>
    </div>
  )
}

const DialogHeader: React.FC<DialogHeaderProps> = ({ children, className }) => {
  return (
    <div className={cn(
      "flex flex-col space-y-1.5 text-center sm:text-left",
      className
    )}>
      {children}
    </div>
  )
}

const DialogTitle: React.FC<DialogTitleProps> = ({ children, className }) => {
  return (
    <h2 className={cn(
      "text-lg font-semibold leading-none tracking-tight",
      className
    )}>
      {children}
    </h2>
  )
}

export {
  Dialog,
  DialogTrigger,
  DialogContent,
  DialogHeader,
  DialogTitle,
} 