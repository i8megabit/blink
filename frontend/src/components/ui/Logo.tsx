import React from 'react';

interface LogoProps {
  size?: 'sm' | 'md' | 'lg' | 'xl';
  variant?: 'default' | 'white' | 'dark';
  className?: string;
  showText?: boolean;
}

const Logo: React.FC<LogoProps> = ({ 
  size = 'md', 
  variant = 'default', 
  className = '',
  showText = true 
}) => {
  const sizeClasses = {
    sm: 'w-16 h-8',
    md: 'w-24 h-10',
    lg: 'w-32 h-12',
    xl: 'w-40 h-16',
  };

  const variantClasses = {
    default: 'text-gray-900',
    white: 'text-white',
    dark: 'text-gray-100',
  };

  const iconSize = {
    sm: 16,
    md: 20,
    lg: 24,
    xl: 32,
  };

  const textSize = {
    sm: 'text-sm',
    md: 'text-base',
    lg: 'text-lg',
    xl: 'text-xl',
  };

  return (
    <div className={`flex items-center gap-2 ${sizeClasses[size]} ${className}`}>
      {/* Иконка */}
      <div className="relative">
        <svg 
          width={iconSize[size]} 
          height={iconSize[size]} 
          viewBox="0 0 24 24" 
          className="flex-shrink-0"
        >
          <defs>
            <linearGradient id="logoGradient" x1="0%" y1="0%" x2="100%" y2="100%">
              <stop offset="0%" style={{ stopColor: '#0ea5e9', stopOpacity: 1 }} />
              <stop offset="100%" style={{ stopColor: '#0284c7', stopOpacity: 1 }} />
            </linearGradient>
            <filter id="logoShadow" x="-20%" y="-20%" width="140%" height="140%">
              <feDropShadow dx="0" dy="1" stdDeviation="2" floodColor="#000000" floodOpacity="0.1"/>
            </filter>
          </defs>
          
          {/* Фоновая форма */}
          <circle 
            cx="12" 
            cy="12" 
            r="10" 
            fill="url(#logoGradient)" 
            filter="url(#logoShadow)"
          />
          
          {/* Иконка ссылки */}
          <g transform="translate(4, 4)">
            <path 
              d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z" 
              fill="white" 
              opacity="0.9"
            />
            <path 
              d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z" 
              fill="white" 
              opacity="0.9"
            />
            <path 
              d="M12 6l-2 2 2 2" 
              stroke="white" 
              strokeWidth="1.5" 
              fill="none" 
              strokeLinecap="round"
            />
          </g>
        </svg>
      </div>

      {/* Текст */}
      {showText && (
        <span className={`font-semibold ${textSize[size]} ${variantClasses[variant]} tracking-tight`}>
          reLink
        </span>
      )}
    </div>
  );
};

export default Logo; 