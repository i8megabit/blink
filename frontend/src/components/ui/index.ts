export { Button } from './Button';
export { Card, CardHeader, CardTitle, CardDescription, CardContent, CardFooter } from './Card';
export { Badge } from './Badge';
export { Input } from './Input';
export { Textarea } from './Textarea';
export { Modal } from './Modal';
export { Progress } from './Progress';
export { Select } from './Select';
export { Table, TableHeader, TableBody, TableRow, TableHead, TableCell } from './Table';
export { Tag } from './Tag'; 