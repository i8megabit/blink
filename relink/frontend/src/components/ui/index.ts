// ===== UI КОМПОНЕНТЫ =====

export { Button } from './Button'
export { Card } from './Card'
export { Input } from './Input'
export { Badge } from './Badge'
export { default as Logo } from './Logo'
export { Progress } from './Progress'

// Re-export types for better DX
export type { ButtonProps } from './Button'
export type { CardProps } from './Card'
export type { InputProps } from './Input'
export type { BadgeProps } from './Badge' 