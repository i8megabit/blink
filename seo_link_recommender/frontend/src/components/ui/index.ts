// UI Components
export { Button } from './Button'
export { Card, CardHeader, CardFooter, CardTitle, CardDescription, CardContent } from './Card'
export { Input } from './Input'
export { Badge } from './Badge'
export { Progress } from './Progress'

// Types
export type { ButtonProps } from './Button'
export type { CardProps, CardHeaderProps, CardFooterProps, CardTitleProps, CardDescriptionProps, CardContentProps } from './Card'
export type { InputProps } from './Input'
export type { BadgeProps } from './Badge'
export type { ProgressProps } from './Progress' 