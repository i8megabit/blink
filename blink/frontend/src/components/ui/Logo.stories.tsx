import type { Meta, StoryObj } from '@storybook/react';
import Logo from './Logo';

const meta: Meta<typeof Logo> = {
  title: 'UI/Logo',
  component: Logo,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    size: {
      control: { type: 'select' },
      options: ['sm', 'md', 'lg', 'xl'],
    },
    variant: {
      control: { type: 'select' },
      options: ['default', 'white', 'dark'],
    },
    showText: {
      control: { type: 'boolean' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

// 🏆 Основные варианты
export const Default: Story = {
  args: {
    size: 'md',
    variant: 'default',
    showText: true,
  },
};

export const Large: Story = {
  args: {
    size: 'lg',
    variant: 'default',
    showText: true,
  },
};

export const IconOnly: Story = {
  args: {
    size: 'md',
    variant: 'default',
    showText: false,
  },
};

// 🎨 Варианты цветов
export const White: Story = {
  args: {
    size: 'md',
    variant: 'white',
    showText: true,
  },
  parameters: {
    backgrounds: { default: 'dark' },
  },
};

export const Dark: Story = {
  args: {
    size: 'md',
    variant: 'dark',
    showText: true,
  },
  parameters: {
    backgrounds: { default: 'dark' },
  },
};

// 📏 Размеры
export const Small: Story = {
  args: {
    size: 'sm',
    variant: 'default',
    showText: true,
  },
};

export const ExtraLarge: Story = {
  args: {
    size: 'xl',
    variant: 'default',
    showText: true,
  },
};

// 🎯 Все варианты
export const AllVariants: Story = {
  render: () => (
    <div className="space-y-8 p-8">
      <div className="space-y-4">
        <h3 className="text-lg font-semibold">Размеры</h3>
        <div className="flex items-center gap-4">
          <Logo size="sm" />
          <Logo size="md" />
          <Logo size="lg" />
          <Logo size="xl" />
        </div>
      </div>
      
      <div className="space-y-4">
        <h3 className="text-lg font-semibold">Варианты цветов</h3>
        <div className="flex items-center gap-4">
          <Logo variant="default" />
          <div className="bg-gray-800 p-2 rounded">
            <Logo variant="white" />
          </div>
          <div className="bg-gray-900 p-2 rounded">
            <Logo variant="dark" />
          </div>
        </div>
      </div>
      
      <div className="space-y-4">
        <h3 className="text-lg font-semibold">Только иконка</h3>
        <div className="flex items-center gap-4">
          <Logo size="sm" showText={false} />
          <Logo size="md" showText={false} />
          <Logo size="lg" showText={false} />
          <Logo size="xl" showText={false} />
        </div>
      </div>
    </div>
  ),
}; 