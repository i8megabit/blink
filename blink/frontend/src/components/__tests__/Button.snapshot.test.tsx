import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { Button } from '../ui/Button';

describe('Button Snapshot', () => {
  it('рендерится с текстом', () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится с иконкой', () => {
    const Icon = () => <span>🔍</span>;
    const { container } = render(<Button icon={<Icon />}>Search</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится в loading состоянии', () => {
    const { container } = render(<Button loading>Loading</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится в disabled состоянии', () => {
    const { container } = render(<Button disabled>Disabled</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится с variant primary', () => {
    const { container } = render(<Button variant="primary">Primary</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится с variant secondary', () => {
    const { container } = render(<Button variant="secondary">Secondary</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится с variant outline', () => {
    const { container } = render(<Button variant="outline">Outline</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится с variant ghost', () => {
    const { container } = render(<Button variant="ghost">Ghost</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится с size sm', () => {
    const { container } = render(<Button size="sm">Small</Button>);
    expect(container).toMatchSnapshot();
  });

  it('рендерится с size lg', () => {
    const { container } = render(<Button size="lg">Large</Button>);
    expect(container).toMatchSnapshot();
  });
}); 