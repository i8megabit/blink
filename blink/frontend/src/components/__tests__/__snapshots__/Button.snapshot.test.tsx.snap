// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è –≤ disabled —Å–æ—Å—Ç–æ—è–Ω–∏–∏ 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
    disabled=""
  >
    Disabled
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è –≤ loading —Å–æ—Å—Ç–æ—è–Ω–∏–∏ 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
    disabled=""
  >
    <svg
      class="mr-2 h-4 w-4 animate-spin"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="opacity-25"
        cx="12"
        cy="12"
        r="10"
        stroke="currentColor"
        stroke-width="4"
      />
      <path
        class="opacity-75"
        d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        fill="currentColor"
      />
    </svg>
    Loading
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è —Å size lg 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-12 px-8 text-lg"
  >
    Large
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è —Å size sm 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-8 px-3 text-xs"
  >
    Small
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è —Å variant ghost 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
  >
    Ghost
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è —Å variant outline 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
  >
    Outline
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è —Å variant primary 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
  >
    Primary
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è —Å variant secondary 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
  >
    Secondary
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è —Å –∏–∫–æ–Ω–∫–æ–π 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
  >
    <span
      class="mr-2"
    >
      <span>
        üîç
      </span>
    </span>
    Search
  </button>
</div>
`;

exports[`Button Snapshot > —Ä–µ–Ω–¥–µ—Ä–∏—Ç—Å—è —Å —Ç–µ–∫—Å—Ç–æ–º 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center rounded-md font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
  >
    Click me
  </button>
</div>
`;
